hazelcast/src/main/java/com/hazelcast/internal/util/XmlUtil.java

```diff
@@ -0,0 +1,218 @@
+            TransformerFactory transformerFactory = getTransformerFactory();
+            /*
+             * Older versions of Xalan still use this method of setting indent values.
+             * Attempt to make this work but don't completely fail if it's a problem.
+             */
+            try {
+                transformerFactory.setAttribute("indent-number", indent);
+            } catch (IllegalArgumentException e) {
+                if (LOGGER.isFinestEnabled()) {
+                    LOGGER.finest("Failed to set indent-number attribute; cause: " + e.getMessage());
+                }
+            }
+            Transformer transformer = transformerFactory.newTransformer();
+            // workaround IBM Java behavior - the silent ignorance of issues during the transformation.
+            transformer.setErrorListener(ThrowingErrorListener.INSTANCE);
+            transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+            transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-8");
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            /*
+             * Newer versions of Xalan will look for a fully-qualified output property in order to specify amount of
+             * indentation to use. Attempt to make this work as well but again don't completely fail if it's a problem.
+             */
+            try {
+                transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", Integer.toString(indent));
+            } catch (IllegalArgumentException e) {
+                if (LOGGER.isFinestEnabled()) {
+                    LOGGER.finest("Failed to set indent-amount property; cause: " + e.getMessage());
+                }
+            }
+            transformer.transform(xmlInput, xmlOutput);
+            return xmlOutput.getWriter().toString();
+        } catch (Exception e) {
+            LOGGER.warning(e);
+            return input;
+        } finally {
+            if (xmlOutput != null) {
+                closeResource(xmlOutput.getWriter());
+            }
+        }
+    }
+
+    /**
+     * Returns ErrorListener implementation which just throws the original error.
+     */
+    public ErrorListener getErrorListener() {
+        return ThrowingErrorListener.INSTANCE;
+    }
+
+    private static void setAttribute(TransformerFactory transformerFactory, String attributeName) {
+        try {
+            transformerFactory.setAttribute(attributeName, "");
+        } catch (IllegalArgumentException iae) {
+            if (Boolean.getBoolean(SYSTEM_PROPERTY_DISABLE_XXE_PROTECTION)) {
+                LOGGER.warning("Enabling XXE protection failed. The attribute " + attributeName
+                        + " is not supported by the TransformerFactory. The " + SYSTEM_PROPERTY_DISABLE_XXE_PROTECTION
+                        + " system property is used so the XML processing continues in the UNSAFE mode!!!");
```

Comment:
Probably we can be explicit here add that unsafe means the XXE protection is disabled, so 
> ... system property is used so the XML processing continues in the UNSAFE mode **with XXE protection disabled** !!!

Also, maybe we can consider using `unsecured` instead of `unsafe` to better describe the nature of the condition.

Same for the below `setProperty()`.